cpu/native/socket_zep/socket_zep.c

```diff
@@ -0,0 +1,424 @@
+                        (zep->length > n) || (zep->chan != dev->netdev.chan) ||
+                        /* TODO promiscous mode */
+                        _dst_not_me(dev, payload)) {
+                        /* TODO: check checksum */
+                        return -1;
+                    }
+                    /* don't hand FCS to stack */
+                    size = zep->length - sizeof(uint16_t);
+                    memcpy(buf, payload, size);
+                    if (info != NULL) {
+                        struct netdev2_radio_rx_info *rx_info = info;
+                        rx_info->lqi = zep->lqi_val;
+                        rx_info->rssi = UINT8_MAX;
+                    }
+                    break;
+                }
+                default:
+                    return -1;
+            }
+        }
+        else if (size == 0) {
+            DEBUG("zep: ignoring null-event\n");
+            return -1;
+        }
+        else if (size == -1) {
+            if ((errno == EAGAIN) || (errno == EWOULDBLOCK)) {
+            }
+            else {
+                err(EXIT_FAILURE, "zep: read");
+            }
+        }
+        else {
+            errx(EXIT_FAILURE, "internal error _rx_event");
+        }
+    }
+    _continue_reading(dev);
+#ifdef MODULE_NETSTATS_L2
+    netdev->stats.rx_count++;
+    netdev->stats.rx_bytes += size;
+#endif
+    return size;
+}
+
+static void _isr(netdev2_t *netdev)
+{
+    if (netdev->event_callback) {
+        socket_zep_t *dev = (socket_zep_t *)netdev;
+
+        netdev->event_callback(netdev, dev->last_event);
+    }
+    return;
+}
+
+static void _socket_isr(int fd, void *arg)
+{
+    (void)fd;
+    (void)arg;
+    netdev2_t *netdev = zep;           /* TODO replace with argument */
+
+    if (zep == NULL) {
+        return;
+    }
+    if (netdev->event_callback) {
+        socket_zep_t *dev = (socket_zep_t *)zep;
+
+        dev->last_event = NETDEV2_EVENT_RX_COMPLETE;
+        netdev->event_callback(netdev, NETDEV2_EVENT_ISR);
+    }
+}
+
+static int _init(netdev2_t *netdev)
+{
+    socket_zep_t *dev = (socket_zep_t *)netdev;
+
+    assert(dev != NULL);
+    assert(zep == NULL);    /* TODO remove */
+    zep = netdev;           /* TODO remove */
+    dev->netdev.chan = IEEE802154_DEFAULT_CHANNEL;
+    dev->netdev.pan = IEEE802154_DEFAULT_PANID;
+#ifdef MODULE_GNRC_SIXLOWPAN
+    dev->netdev.proto = GNRC_NETTYPE_SIXLOWPAN;
+#elif MODULE_GNRC
+    dev->netdev.proto = GNRC_NETTYPE_UNDEF;
+#endif
+    dev->seq = random_uint32();
+
+    return 0;
+}
+
+static int _get(netdev2_t *netdev, netopt_t opt, void *value, size_t max_len)
+{
+    socket_zep_t *dev = (socket_zep_t *)netdev;
+    uint16_t *v = value;
+
+    assert(dev != NULL);
+    switch (opt) {
+        case NETOPT_MAX_PACKET_SIZE:
+            if (max_len != sizeof(uint16_t)) {
+                return -EOVERFLOW;
+            }
+            *v = SOCKET_ZEP_FRAME_PAYLOAD_LEN;
```

Comment:
Done in https://github.com/RIOT-OS/RIOT/pull/6121/files#diff-01bcdc6a0bf2aa346059e749a7178abaR309 btw ;-)